extras/linter_install_tutorial.md

```diff
@@ -0,0 +1,96 @@
+
+## About
+
+This section describes how to set up running a liner uploading code onto a GitHub repository (git commit ... -> git push).
+A [linter](https://en.wikipedia.org/wiki/Lint_(software)) is a code analysis tool designed to analyze code and flag for programming errors, bugs and stylistic errors.
+This tool allows one to drastically improve their productivity when writing code for their research projects as stylistic errors and programming errors will automatically be covered through this set-up.
+Disclaimer: The following instructions only work for python code as R doesn't have a working linter. (Subject to change if new information were to come out on this fact.)
+
+## Programs and Packages needed to set up Pipeline
+
+Make sure [conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/download.html) or another python package manager is installed on your computer.
+This pipeline is designed to run on python >=3.5.
+
+This setup will also use the following packages (versions can be changed as needed):
+1. pre-commit==2.9.3
+2. black==20.8b1 
+3. jupytext==1.9.1
+
+## Installation Instructions
+
+1. Set up your python environment to use python>=3.5 
+2. Call ``pip install pre-commit==2.9.3``
+3. call ``pip install black==20.8b1``
+4. Create a file with the extension .pre-commit-config.yaml and fill in with lines of code reproduced in the [Configuration Files](#Pre-commit) section
+5. Create a file with the extension .flake8 and fill in with lines of code reproduced in the [Configuration Files](#Flake8) section
+6. Create a file with the extension .toml and fill in with lines of code reproduced in the [Configuration Files](#Black) section
+7. Then call command ``pre-commit install`` to install the pipeline to run after the ``git commit`` command is called.
+8. Now the pipeline is installed, and you can test it by calling ``pre-commit``.
+9. Now you can proceed with normal code development and GitHub processes
+10. When you call `git commit`, you will see a few checks being at the start of the committing process.
+11. Checkpoints will fail as files are modified or flake8 errors persist
+12. Review that changes files are to one's liking or manually make changes
+13. Lastly, call ``git add`` on the changed files then call the same ``git commit`` command again.
+
+## Warnings
+
+This pipeline is designed to run black on jupyter notebooks and does not run on individual python scripts.
+The flake8 step will complain about non-formatted python scripts as no changes would have been made at that point.
+Simply run the command `black <python script>` and re-add the changed file.
+
+## Configuration Files 
+
+### Pre-commit
+
+```yaml
+repos:
+
+- repo: https://github.com/mwouts/jupytext
+  rev: 41b7e9c93b50da1d6feb486aceb2c4d534374090
+  hooks:
+  - id: jupytext
+    name: jupytext_auto_linter
+    args: [--set-formats, 'ipynb,py', --from, ipynb, --pipe, black, --to, py:light, --sync]
```

Comment:
Is there anyway to change where the sync'd .py file is stored? By default its in the same location as the notebook

Reply:
There is but the only problem is that all notebooks will be gathered into a single location and all scripts will do the same. Depending on organization some may or may not want to enable that feature; however, I will update my instructions file to enable that feature in cases one wants to do that. Great question!